app/helpers/reservations_helper.rb

```diff
@@ -41,4 +41,10 @@ def check_schedule(schedules, day, time)
       Time.parse(time).between?(Time.parse(select_day[0].start_time.strftime("%H:%M").to_s), Time.parse(select_day[0].end_time.strftime("%H:%M").to_s))
     end
   end
+
+  def check_date_before_today(day)
+    if day < Date.today
+      return true
+    end
```

Comment:
この条件に当てはまらない場合は nil が返ると思います。
シンプルに以下で良さそうに思いました。基本的に返り値は true or false と型を合わせてあげたほうがよさそう。

```suggestion
    day < Date.today
```

Reply:
盲点でした...ありがとうございます！